### PR TITLE
fix: don't package projects with no parts

### PR DESCRIPTION
**Pull request by cmatsuoka**
Skip package creation When a project has no parts defined.

- [ ] Have you followed the guidelines for contributing?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `make lint && make test`?
- [ ] Have you added an entry to the changelog (`docs/reference/changelog.rst`)?

---
CRAFT-4693

### CODE REVIEW

**Review by lengau — APPROVED** [review]
This is an interesting edge case. I'm fine with it either behaving like this or packing an empty project.

File: craft_application/commands/lifecycle.py [hunk 1]
```diff
@@ -485,16 +485,20 @@ def _run_real(
...
+            emit.progress("Packing...")
+            try:
+                packages = self._services.package.pack(
+                    self.services.get("lifecycle").prime_dir, parsed_args.output
+                )
+            except Exception as err:
+                if debug:
+                    emit.progress(str(err), permanent=True)
+                    _launch_shell()
+                raise
```
Comment (mr-cal) [inline craft_application/commands/lifecycle.py]: nit:

```python
if a:
  b
else:
  c
```

is more readable than:

```python
if not a:
  c
else:
  b
```

**Review by mr-cal — APPROVED** [review]
I agree with Alex - I could go either way.

File: craft_application/commands/lifecycle.py (hunk 1 above)
Comment (cmatsuoka) [reply craft_application/commands/lifecycle.py]: Fixing.
